src/components/MotionGenerator.tsx

```diff
@@ -1,19 +1,26 @@
 "use client";
 import { motion } from "@/types/motion";
 import motions from "@/data/motion.json";
-import { useEffect, useState } from "react";
+import { useContext, useEffect, useState } from "react";
 import { IconInfo } from "@/components/icons/info";
 import { useLang } from "@/lib/useLang";
 import { GenericButton } from "./GenericButton";
-import { IconRotateCCW } from "./icons/RotateCCW";
+import { LinkButton } from "./LinkButton";
+import { DebateContext } from "@/contexts/DebateContext";
+import { IconDice } from "./icons/Dice";
+import { IconClock } from "./icons/Clock";
 
 const MotionGenerator = () => {
   const [motion, setMotion] = useState<motion | null>(null);
   const infoslideString = useLang("infoslide");
+  const debateSetupPath = "/../oxford-debate/setup";
+  const debateContext = useContext(DebateContext);
 
   function generateMotion(): motion {
     // TODO: add option to only generate motions in a chosen language
-    return motions[Math.floor(Math.random() * motions.length)];
+    const randomMotion = motions[Math.floor(Math.random() * motions.length)];
+    debateContext.setConf({ ...debateContext.conf, motion: randomMotion.motion || "" });
+    return randomMotion;
```

Comment:
This change overwrites the motion even if you have no desire to use it, and it does so on every page load. I'd revert.

I think you might be able to use the onClick on a link button and have it work. (?)

Reply:
You're right. I'm unsure if the `onClick` will work since the link button redirects the user to another page, but I will check that. Either way, the current behavior is undesirable, so will fix it.

Reply:
AFAIK the default anchor behaviour is dependent on value returned by the onclick?
it's kinda funky